frappe/utils/nestedset.py

```diff
@@ -137,7 +137,8 @@ def update_move_node(doc, parent_field):
 	frappe.db.sql("""update `tab{0}` set lft = -lft + %s, rgt = -rgt + %s, modified=%s
 		where lft < 0""".format(doc.doctype), (new_diff, new_diff, n))
 
-def rebuild_tree(doctype, parent_field):
+@frappe.whitelist()
+def rebuild_tree(doctype, parent_field, commit=0):
```

Comment:
This should be only allowed for **Administrator** or **System Manager**

Reply:
@surajshetty3416 this function is also called when merging, adding and deleting nodes. So added a permission check for doctype, should be enough, right?

Reply:
@deepeshgarg007 You can check for permission only if this function is called via API request (or clientside)

You can [refer to this](https://github.com/frappe/frappe/blob/develop/frappe/core/doctype/user_permission/user_permission.py#L60) for implementation.